org.jacoco.ant.test/src/org/jacoco/ant/IllegalReflectiveAccessTarget.java

```diff
@@ -0,0 +1,46 @@
+/*******************************************************************************
+ * Copyright (c) 2009, 2022 Mountainminds GmbH & Co. KG and Contributors
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    Evgeny Mandrikov - initial API and implementation
+ *
+ *******************************************************************************/
+
+package org.jacoco.ant;
+
+import java.lang.reflect.Constructor;
+
+public class IllegalReflectiveAccessTarget {
+
+	public static void main(String[] args) throws Exception {
+		try {
+			Class.forName("java.lang.invoke.MethodHandles$Lookup")
+					.getField("ORIGINAL");
+		} catch (ClassNotFoundException e) {
+			// Java < 7
+			return;
+		} catch (NoSuchFieldException e) {
+			// Java < 16
+			return;
+		}
```

Comment:
```suggestion
		try {
			Class.forName("java.net.UnixDomainSocketAddress");
		} catch (ClassNotFoundException e) {
			// Java < 16
			return;
		}
```

Java 16 detection can be simplified with checking for `UnixDomainSocketAddress` which was introduced with this release.